pkg/logicalbackup/logicalbackup.go

```diff
@@ -548,6 +484,72 @@ func (b *LogicalBackup) initPublication(conn *pgx.Conn) error {
 	return nil
 }
 
+// register tables for the backup; add replica identity when necessary
+func (b *LogicalBackup) prepareTablesForPublication(conn *pgx.Conn) error {
+	// fetch all tables from the current publication, together with the information on whether we need to create
+	// replica identitiy full for them
+	rows, err := conn.Query(`
+			select c.oid,
+				   n.nspname,
+				   c.relname,
+       			   CASE
+         			WHEN csr.oid IS NULL AND c.relreplident IN ('d', 'n') THEN true
+         			ELSE false END AS has_replica_identity
+			from pg_class c
+				   join pg_namespace n on n.oid = c.relnamespace
+       			   join pg_publication_tables pub on (c.relname = pub.tablename and n.nspname = pub.schemaname)
+       			   left join pg_constraint csr on (csr.conrelid = c.oid and csr.contype = 'p')
+			where c.relkind = 'r'
+  			  and pub.pubname = $1
+			order by c.oid;`, b.cfg.PublicationName)
+	if err != nil {
+		return fmt.Errorf("could not execute query: %v", err)
+	}
+
+	tables := make([]struct {
```

Comment:
why not declare local `type` inside the function?

Reply:
Because `tables` scope is broader than just a nested function, which could be made to return the tables, but then we would just move the declaration to the function return type rather than doing this in a straightforward manner in the outside function.